### ISSUE TITLE
Craft 5 support for Matrix

### ISSUE THREAD

**Issue by niektenhoopen** [simplygoodwork/craft-block-usage#10]
Version 2.0.0 is currently tagged as Craft 5 compatible but does not work well with the new Matrix blocks, they are interpreted as Entry Types which is technically correct, but the plug-in seems to handle this as a Craft-4-style entry type (part of a section).

What are your plans for making this plug-in compatible with Craft v5?

**Comment by andrewmenich** [comment]
@niektenhoopen I noticed a few issues with Matrix handling in Craft 5. I just released [v2.0.1](https://github.com/simplygoodwork/craft-block-usage/releases/tag/v2.0.1) that fixed those. It seems to be working as we intended now but if you've got a different use case or expectation in mind let us know (screenshots or examples would be helpful!) We're always open to improving things. Thanks!

**Comment by niektenhoopen** [comment]
@andrewmenich It's not working as intended right now. See the screenshot below. Left is Craft v4, right is Craft v5. I am still trying to figure out what is happening.

<img width="915" alt="image" src="https://github.com/user-attachments/assets/e254c36e-cae9-4551-9429-71a859b88aaa">

**Comment by andrewmenich** [comment]
@niektenhoopen Thanks for the feedback. There is still an issue with the count on the field layout usage on the Field page, but if you click into one of those, the resulting page should present a more accurate reflection of their actual usage. I'll get this counting issue fixed this week. Please note that currently these counts and usage reports are including drafts as well, so that might be something else we need to look into.

**Comment by niektenhoopen** [comment]
@andrewmenich No, if I click into a field it shows the same behavior. It shows far too many results, I think it may be revisions or something.

I have created PR #12. This gives the correct results, but the index is still very slow. It's faster when you show the details of a specific block. The common factor was that they all had the same canonical ID so I decided to check for the uniqueness of that field.
